pysparkling/sql/expressions/mappers.py

```diff
@@ -75,6 +87,115 @@ def __str__(self):
         return "({0} IS NOT NULL)".format(self.column)
 
 
+class Cast(Expression):
+    def __init__(self, e, dtype):
+        super().__init__(e, dtype)
+
+        self.e = e
+        if not isinstance(dtype, str):
+            try:
+                dtype = dtype.typeName()
+            except AttributeError:
+                pass
+        self.dtype = dtype
```

Comment:
Isn't this overridden no matter what right after?

Reply:
It does not seem useful to store dtype_str too

Reply:
True. I'll remove that.